src/H5.c

```diff
@@ -224,50 +230,53 @@ H5_init_library(void)
     } /* end if */
 
     /*
-     * Initialize interfaces that might not be able to initialize themselves
-     * soon enough.  The file & dataset interfaces must be initialized because
-     * calling H5P_create() might require the file/dataset property classes to be
-     * initialized.  The property interface must be initialized before the file
-     * & dataset interfaces though, in order to provide them with the proper
-     * property classes.
-     * The link interface needs to be initialized so that link property lists
-     * have their properties registered.
+     * Initialize interfaces that use macros of the form "(H5OPEN <var>)", so
+     * that the variable returned through the macros nas been initialized.
+     * Also initialize some interfaces that might not be able to initialize
+     * themselves soon enough.
+     *
+     * Interfaces returning variables through a macro: H5E, H5FD, H5O, H5P, H5T
+     *
+     * The link interface needs to be initialized so that the external link
+     *   class is registered.
+     *
      * The FS module needs to be initialized as a result of the fix for HDFFV-10160:
      *   It might not be initialized during normal file open.
      *   When the application does not close the file, routines in the module might
      *   be called via H5_term_library() when shutting down the file.
+     *
      * The dataspace interface needs to be initialized so that future IDs for
      *   dataspaces work.
+     *
+     * The VFD & VOL interfaces need to be initialized before the H5P interface
+     *   so that the default VFD and default VOL connector are ready for the
+     *   default FAPL.
+     *
      */
-    {
-        /* clang-format off */
-        struct {
-            herr_t (*func)(void);
-            const char *descr;
-        } initializer[] = {
-            {H5E_init, "error"}
-        ,   {H5VL_init_phase1, "VOL"}
-        ,   {H5SL_init, "skip lists"}
-        ,   {H5FD_init, "VFD"}
-        ,   {H5_default_vfd_init, "default VFD"}
-        ,   {H5P_init_phase1, "property list"}
-        ,   {H5AC_init, "metadata caching"}
-        ,   {H5L_init, "link"}
-        ,   {H5S_init, "dataspace"}
-        ,   {H5PL_init, "plugins"}
-        /* Finish initializing interfaces that depend on the interfaces above */
-        ,   {H5P_init_phase2, "property list"}
-        ,   {H5VL_init_phase2, "VOL"}
-        };
-
-        for (i = 0; i < NELMTS(initializer); i++) {
-            if (initializer[i].func() < 0) {
-                HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL,
-                    "unable to initialize %s interface", initializer[i].descr);
-            }
-        }
-        /* clang-format on */
-    }
+    if (H5E_init() < 0)
```

Comment:
Went back to the original initialization code, which is simpler and easier to parse.